src/components/screens/FormScreen.tsx

```diff
@@ -133,27 +131,45 @@ function randomizer(
     const ingredient = key as keyof PaniniFormData;
     const ingredientType = key as FormDataType;
 
-    if (["bread", "sandwichName", "serving"].indexOf(ingredient) >= 0) {
-      randomForm[ingredient] = [randomValue(ingredientType)];
-    } else if (["cutlery", "napkins", "topping"].indexOf(ingredient) >= 0) {
-      const isTrue = Math.random() >= 0.5;
-      randomForm[ingredient] = isTrue ? [randomValue(ingredientType)] : [];
-    } else if (["spreads", "vegetables"].indexOf(ingredient) >= 0) {
-      const dataArray = data(ingredientType);
-      for (let index = 0; index < dataArray.length; index++) {
-        const isTrue = Math.random() >= 0.5;
-        randomForm[ingredient][index] = isTrue ? dataArray[index] : "";
-      }
-    } else {
-      const randomNumber = Math.ceil(Math.random() * 5);
-      for (let index = 0; index < randomNumber; index++) {
-        randomForm[ingredient][index] = randomValue(ingredientType);
-      }
+    switch (ingredientType) {
+      //single-value cases
+      case FormDataType.Bread:
+      case FormDataType.Name:
+      case FormDataType.Serving:
+        randomForm[ingredient] = [randomValue(ingredientType)];
+        break;
+      //booleable cases
+      case FormDataType.Cutlery:
+      case FormDataType.Napkins:
+      case FormDataType.Topping:
+        {
+          const isTrue = Math.random() >= 0.5;
+          randomForm[ingredient] = isTrue ? [randomValue(ingredientType)] : [];
+        }
+        break;
+      //non-repeating cases
+      case FormDataType.Spreads:
+      case FormDataType.Vegetables:
+        {
+          const dataArray = data(ingredientType);
+          for (let index = 0; index < dataArray.length; index++) {
+            const isTrue = Math.random() >= 0.5;
+            randomForm[ingredient][index] = isTrue ? dataArray[index] : "";
+          }
+        }
+        break;
+      //all remaining cases
+      default:
+        {
+          const randomNumber = Math.ceil(Math.random() * 4);
+          for (let index = 0; index < randomNumber; index++) {
+            randomForm[ingredient][index] = randomValue(ingredientType);
+          }
+        }
+        break;
     }
   }
-
-  formMethods.reset(randomForm);
-  return;
+  return randomForm;
```

Comment:
The function returns the randomForm object now, allowing previous line 68 to be inverted, so to speak. Now im putting randomForm into formMethods, not the other way around.